extensions/amp-story/1.0/amp-story-store-service.js

```diff
@@ -436,6 +441,15 @@ const actions = (state, action, data) => {
         ...state,
         [StateProperty.VIEWER_CUSTOM_CONTROLS]: data,
       });
+    case Action.TOGGLE_SUBSCRIPTION_DIALOG_IS_VISIBLE:
+      // Don't change the state if the new state is equal to the current state.
+      if (state[StateProperty.SUBSCRIPTION_DIALOG_IS_VISIBLE_STATE] === data) {
+        return state;
+      }
+      return /** @type {!State} */ ({
+        ...state,
+        [StateProperty.SUBSCRIPTION_DIALOG_IS_VISIBLE_STATE]: !!data,
```

Comment:
You also need to pause the Story. Here's how the amp-access implementation used to do it:

```
    // Triggers the amp-acess paywall.
    case Action.TOGGLE_ACCESS:
      // Don't change the PAUSED_STATE if ACCESS_STATE is not changed.
      if (state[StateProperty.ACCESS_STATE] === data) {
        return state;
      }

      return /** @type {!State} */ ({
        ...state,
        [StateProperty.ACCESS_STATE]: !!data,
        [StateProperty.PAUSED_STATE]: !!data,
      });
```

Reply:
Done. Interesting I did that by explicitly pausing the page in amp-story. This does look better.